sdk/storage/Azure.Storage.Files.Shares/perf/Microsoft.Azure.Storage.File.Perf/Scenarios/DownloadFile.cs

```diff
@@ -0,0 +1,94 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Test.Perf;
+
+namespace Microsoft.Azure.Storage.File.Perf.Scenarios
+{
+    /// <summary>
+    /// The performance test scenario focused on downloading files from the Microsoft Azure File Shares storage.
+    /// </summary>
+    /// <seealso cref="Azure.Test.Perf.PerfTest{SizeOptions}" />
+    public sealed class DownloadFile : PerfTest<SizeOptions>
+    {
+        /// <summary>
+        /// Reference to the Share used by the test in the Microsoft Azure File service.
+        /// </summary>
+        private CloudFileShare _cloudFileShare;
+
+        /// <summary>
+        /// Reference to the file used by the test in the Microsoft Azure File service.
+        /// </summary>
+        private CloudFile _cloudFile;
+
+        /// <summary>
+        /// Local stream uploaded to Microsoft Azure File service.
+        /// </summary>
+        private readonly Stream _stream;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DownloadFile"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        public DownloadFile(SizeOptions options) : base(options)
+        {
+            _stream = RandomStream.Create(options.Size);
+        }
+
+        public override void Dispose(bool disposing)
+        {
+            _stream.Dispose();
+            base.Dispose(disposing);
+        }
+
+        /// <summary>
+        /// Creates a cloud file share to be used by the test.
+        /// Also, creates a file reference in the cloud file share and uploads data stream to the cloud file.
+        /// </summary>
+        /// <returns></returns>
+        public override async Task SetupAsync()
```

Comment:
Should this be the global setup?   That would allow us to create a single file that gets used across all parallel instances of the scenario, which saves us from potentially having to upload the larger-size assets over and over.

Reply:
There was a bit of back and forth about this in the last PR. We want to -
1. Not waste resources on the client and backend
2. Model the operations as we expect users to be using them
3. Not artificially hamper or boost the performance with the usage pattern
This is what I ended up with in the T2 test.

Making this a global setup operation would mean the test is downloading from the same storage resource across all instances.

I can make the change and I'll also update the T2 tests to maintain parity.

Reply:
There was also a lengthy verbal discussion for the approach in the context of Data Lake, where the recommendation was to prefer a single download across all instances because "it shouldn't matter to the service."    _([ref](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/perf/Azure.Storage.Files.DataLake.Perf/Scenarios/Read.cs#L70))_    

Personally, I don't have a strong opinion here, but @mikeharder did.   I'm good either way, but we should probably use the same approach across libraries and languages unless we have a reason to deviate... which maybe we do for Files?

Reply:
I don't think we should deviate and I don't have a reason to. I will make the change.

Reply:
I tested out with the two variations and there is a perf difference of 2x to 20x. The results are substantially better when using test specific client references.
We should do the same comparison with Datalake tests.